.net sdk 1.4.1 r4elease notes updated (#4648)
harness/developer-hub: release-notes/feature-flags.md

### .NET SDK

#### Version 1.4.1 

>  - The thread safety metrics issue have been resolved.

Reviewer: Suggestion: what thread safety metrics issue? :) It might be worth explicitly saying a thread safety issue when sending metrics was resolved

 - We no longer store duplicate targets used by metrics in memory. 

>  - Can send global target identifier in the evaluation metrics.

Reviewer: ```suggestion
 - Uses the global target identifier for evaluation metrics.
```
It's not optional behaviour
